File: 27_Nth Fibonacci/nthFibonacci.java
import java.io.*;
import java.lang.*;
import java.util.*;

class Main{
  //This function returns the Nth fibonacci number.
  public static int nthFibonacci(int n){
    if (n == 0) return 0;
    if (n == 1) return 1;
    return nthFibonacci(n-1) + nthFibonacci(n-2);
  }
  //Main Function
  public static void main(String[] args){
    int n = 4;
    System.out.println(nthFibonacci(n));
  }
}

File: 27_Nth Fibonacci/Ideal Solutions/nthFibonacci.py
# Finds Nth Fibonacci Number
def nthFibonacci(Number):
    first, second = 0, 1

    # Base Case
    if Number < 0:
        print("Wrong Input")
    elif Number == 0:
        return first
    elif Number == 1:
        return second


    # Finding Nth Fibonacci using recurring addition
    else:
        for i in range(2, Number + 1):
            third = first + second
            first = second
            second = third

        return second


if __name__ == "__main__":
    array = []

    # Static Input
    N = 9

    print(nthFibonacci(N))
